pandas/tests/test_generic.py

```diff
@@ -354,6 +354,52 @@ def test_head_tail(self):
             self._compare(o.head(-3), o.head(7))
             self._compare(o.tail(-3), o.tail(7))
 
+    def test_sample(self):
+        o = self._construct(shape=10)
+        
+        # Run for all index types. 
```

Comment:
Does any code here actually depend on the index type?

If not, I would skip these tests here. The `take` method should already be tested directly for all these different index types.